### ISSUE TITLE
Planning implementation of new UI designs (work in progress!) 

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#4922]
(this issue is not ready yet, as we're still finalizing UI design work!)

Coming out of a range of UI analysis and design projects from the past 6 months, Public Lab is now ready to begin implementing new designs across major pages of our PublicLab.org collaborative platform. This project will build on the Bootstrap 4 UI library and our upcoming Style Guide to implement new HTML templates within the Ruby on Rails website at PublicLab.org, focused on the following pages:

* [ ] Dashboard
* [x] People page
* [x] Tags/topics page
* [x] Individual tag pages
* [ ] Questions page
* [x] Profile page

The design work to date can be found in a series of mockups listed on this page: https://publiclab.org/tag/ui-designs.

**Comment by jywarren** [comment]
Very tentatively starting to create a checklist: 

This project includes a variety of tasks, many of which are simpler, making them a good place to get started. These include simpler pages like the People page, the Tags/Topics page, and parts of other pages. These will help you get oriented before taking on a larger project. 

We would be happy with any subset of these tasks being completed, and if you finish them all, we can definitely come up with more to keep you challenged!

Specific tasks include (divided among the different page designs listed above):



* Creating an HTML version of the “card” designs shown in several of the UI mockups

* Testing these at different screen widths and on different devices

* Modifying the People page to show cards of people instead of rows

* adding tools to a less obtrusive popover menu using Bootstrap UI components

* Modifying the People page to show a map on the right side

* Modifying the Tags/Topics page to show cards instead of table rows

* Modifying the Questions page to display an introductory text area

* Creating a secondary Dashboard template to shadow the first, and getting initial topics to appear on it

* Adding a “new topics” area to the dashboard page

* Adding a map of recent nearby content (larger project) to the dashboard page

* Displaying topics on the dashboard based on subscriptions

* Connecting “close card” button to “unsubscribe” function

* Adding “recent and trending tags” section to dashboard

* creating template partials from people and topic card designs for use in other parts of the site

* displaying active topics on the profile page

* restructuring the profile page column layout to follow the new UI design

* compressing profile page tools into a smaller fold-away interface design

* integrating a map of recent content onto the topics page

* developing a map popup interface to display individual recently posted pieces of content on a map



Stretch goals

* [ ] closer integration with new layers UI from Leaflet Environmental Layers
* [ ] search for locations box within maps (as shown in UI designs)
* [ ] integration of alerts to guide subscriptions and welcome to new users

**Comment by karunagoyalk** [comment]
Hello!
 I would like to work on this issue! Please help me to get started

**Comment by jywarren** [comment]
Hi, actually we aren't quite ready for this to be implemented yet, but we
will be soon! Can you check out some other issues for now? Thank you!

**Comment by karunagoyalk** [comment]
@jywarren No problem!
Do you have any other issues for this project or I have to look for some different project

**Comment by jywarren** [comment]
You can check at https://code.publiclab.org/#r=all -- thanks! And some
issues may be claimed, but not worked on recently - you can ask if people
are OK with you jumping in to take them up!

**Comment by jywarren** [comment]
We will probably be starting with the People page redesign at https://publiclab.org/people -- following the mockup at the bottom of: https://publiclab.org/notes/edie_blues/01-16-2019/what-do-you-think-about-this-design-for-the-people-page

However we are going to wait on the Bootstrap 4 upgrade by @Souravirus in #3937 -- thanks!

**Comment by gautamig54** [comment]
Hello @jywarren!! Looking forward to contributing to this issue!

**Comment by ananyaarun** [comment]
Hey, @jywarren ,So at the moment i notice a lot of scope for UI/UX improvements/faults on the Public Labs page.
To name a few :-
1) Cluttered arrangement of text and buttons
2) Lack of color consistency among various buttons
3) Certain pop up messages going outside screen
4) Certain messages beginning with a capital letter while certain do not
5) Missing pop up messages for some buttons
Since these UI changes require discussion and debate, can i go ahead and open individual issues ? Or can we discuss some improvements and implementations here first ?

**Comment by jywarren** [comment]
Yes, perhaps you can make an issue collecting up in a list a lot of
detailed ones? But be aware that some might be resolved if we completely
revamp the relevant page in this big redesign. Still, several may make good
issues for new contributors! Have you made a first-timers-only issue, or
solved on yourself? https://code.publiclab.org/#r=all

**Comment by ananyaarun** [comment]
@jywarren , Sure thats a good idea , perhaps i can take a particular section of the webpage at a time , for example the dashboard, about us sections etc and give a detailed description of the UI/UX changes possible/needed. This can further be discussed on the issue thread and once finalized a PR can be made :) 

Regarding FTO issues , i have opened and solved quite a few already. 
https://github.com/publiclab/plots2/issues/created_by/ananyaarun

Yes, as you suggested certain easy changes, i shall open as FTO issues in the repo.
Thanks for your feedback !

**Comment by jywarren** [comment]
:-) awesome. Thanks so much! And indeed, it'd be great to start copying and
dividing up the specific steps it would take to implement the designs --
making checklists!

**Comment by karunagoyalk** [comment]
@gauravano Hey! can you assign me any issue for this project!

**Comment by gauravano** [comment]
Hey @karunagoyalk, this project is to be solved in Outreachy so please take look at other issues of  https://github.com/publiclab/plots2/issues/ . 

Thanks!

**Comment by vaarigupta** [comment]
hello @jywarren  I am an outreachy applicant and excited  to work on it. How can I start contributing ?

**Comment by jywarren** [comment]
I'm going to close this as we've completed or broken out all tasks! Thanks, all!
